tests/neg/i3540.scala

```diff
@@ -0,0 +1,3 @@
+object Test {
+  implicit class Foo(i: Int, s: String) // error: Implicit classes must accept exactly one primary constructor parameter
+}
```

Comment:
Can you add more tests here:
```scala
implicit class Foo0
implicit class Foo1()
implicit class Foo2()(x: Int)
```

Reply:
You should also add 
```scala
implicit case class Bar0
implicit case class Bar1()
implicit case class Bar2()(x: Int)
```

Reply:
OK. Adding these tests,

Reply:
Probably

Reply:
You simply need to add `// error` next to the lines that do not compile

Reply:
👍 Added.

Reply:
Can you also add the error messages emitted by the compiler:
```scala
implicit class Foo0 // error: <the error message here>
```
You can get the error messages by running in sbt:
```
dotc tests/neg/i3540.scala
```

Reply:
```

Reply:
Added error messages to the comments.